tests/admin_scripts/tests.py

```diff
@@ -1867,6 +1867,21 @@ def _test(self, args, option_b="'2'"):
         )
 
 
+class ExecuteFromCommandLine(SimpleTestCase):
+    """Tests for `execute_from_command_line` itself."""
+
+    def test_prog_name_from_argv0(self):
+        """Program name is computed once from argv argument, not sys.argv (#32177)."""
+        # `python -m django shell -c'1+1'` doesn't have any visible side effects
+        args = ['shell', '-c', '1+1']
+        # sys.argv is guaranteed to be a non-empty list containing strings.
+        # But if execute_from_command_line is correctly just computing prog_name
+        # from the argv argument, violating this invariant shouldn't cause the
+        # TypeError that it used to.
+        with mock.patch.object(sys, 'argv', [None] + args):
+            execute_from_command_line([''] + args)
```

Comment:
It's always better to assert with an excepted value, not only check if a test doesn't crash, so I switched to `django-admin help shell`.